### ISSUE TITLE
wiremock-jre8-standalone affected by high severity CVE due to jetty 9.4.46.v20220331

### ISSUE THREAD

**Issue by txedo** [wiremock/wiremock#1944]
Jar: wiremock-jre8-standalone
Affected version <=2.33.2
CVE: [CVE-2022-2048](https://nvd.nist.gov/vuln/detail/CVE-2022-2048)
Severity: High

[Bumping jetty](https://github.com/wiremock/wiremock/blob/2.33.2/build.gradle#L29) either to [9.4.47.v20220610](https://mvnrepository.com/artifact/org.eclipse.jetty/jetty-server/9.4.47.v20220610) or [9.4.48.v20220622](https://mvnrepository.com/artifact/org.eclipse.jetty/jetty-server/9.4.48.v20220622) should fix the issue by releasing version 2.33.3

# Security scanner output

```
+----------------+-----------+------+--------------------------------------+---------------------+-----------------------------------+-------------+------------+------------+----------------------------------------------------+
|      CVE       | SEVERITY  | CVSS |               PACKAGE                |       VERSION       |              STATUS               |  PUBLISHED  | DISCOVERED | GRACE DAYS |                    DESCRIPTION                     |
+----------------+-----------+------+--------------------------------------+---------------------+-----------------------------------+-------------+------------+------------+----------------------------------------------------+
| CVE-2022-2048  | high      | 7.50 | org.eclipse.jetty_jetty-io           | 9.4.46              | fixed in 11.0.9, 10.0.9, 9.4.47   | 59 days     | < 1 hour   | -30        | In Eclipse Jetty HTTP/2 server implementation,     |
|                |           |      |                                      |                     | 59 days ago                       |             |            |            | when encountering an invalid HTTP/2 request, the   |
|                |           |      |                                      |                     |                                   |             |            |            | error handling has a bug that can wind up not      |
|                |           |      |                                      |                     |                                   |             |            |            | properly...                                        |
+----------------+-----------+------+--------------------------------------+---------------------+-----------------------------------+-------------+------------+------------+----------------------------------------------------+
| CVE-2022-2048  | high      | 7.00 | org.eclipse.jetty.http2_http2-server | 9.4.46              | fixed in 11.0.10, 10.0.10, 9.4.47 | 59 days     | < 1 hour   | -30        | In Eclipse Jetty HTTP/2 server implementation,     |
|                |           |      |                                      |                     | 59 days ago                       |             |            |            | when encountering an invalid HTTP/2 request, the   |
|                |           |      |                                      |                     |                                   |             |            |            | error handling has a bug that can wind up not      |
|                |           |      |                                      |                     |                                   |             |            |            | properly...                                        |
+----------------+-----------+------+--------------------------------------+---------------------+-----------------------------------+-------------+------------+------------+----------------------------------------------------+
```

**Comment by tompahoward** [comment]
relates to #1906 and #1907

**Comment by tompahoward** [comment]
Fixed by https://github.com/wiremock/wiremock/commit/bbbc03f5a0ebcf2377cc5e33e2c067214dc1b06b

**Comment by eschulma** [comment]
This is a problem for us also. When I attempt to force override the Jetty dependency to either of the two newer versions (9.4.47 or 9.4.48) I get errors in WireMock. Specifically when stubbing for requests (/bgcdirectpost.aspx is the desired endpoint) I get the following error:

```
13:51:05.755 [qtp1414967210-25] WARN  (HttpChannel.java:776) - /bgcdirectpost.aspx
java.lang.UnsupportedOperationException: Use .getMatched(String) instead
	at org.eclipse.jetty.http.pathmap.PathMappings.getMatch(PathMappings.java:259)
	at org.eclipse.jetty.servlet.ServletHandler.getMappedServlet(ServletHandler.java:572)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:468)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:763)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:748)
```
 I can give more details if desired.

**Comment by tomakehurst** [comment]
Fixed in 5770119cf0e6b8eeaef2a70ce733eb4f828306e5 and released in 2.34.0
